src/c_api/c_api.cc

```diff
@@ -163,7 +163,7 @@ XGB_DLL int XGDMatrixCreateFromCSCEx(const size_t* col_ptr,
                                      const unsigned* indices,
                                      const bst_float* data,
                                      size_t nindptr,
-                                     size_t nelem,
+                                     size_t,
```

Comment:
Please keep this parameter for now.  I'm not sure about these old APIs.

Reply:
In a previous PR of mine I already left some parameters as unnamed in the C api, should I revert those as well?

Another option would be to use `__attribute__((unused))`, which should work on both GCC and Clang.

Reply:
Got it.  Then let's keep the PR as it's.  Thanks for the good work on reducing warnings!